### PR TITLE
Use slices of items to clean up after tests

### LINKED ISSUE #27582: node e2e test teardown crashes with nil deref if node services could not be started
In `/test/e2e_node/e2e_node_suite_test.go`, the node services are started in the `BeforeSuite` logic. If they fail to start, ginkgo's `Fail` function is called, which ultimately initiates test tear down and thus the `AfterSuite` logic in the same file.

The `AfterSuite` logic calls `e2es.stop()` (defined in `test/e2e_node/e2e_service.go`), which will try to e.g. `es.stopService(es.kubeletCmd)`, where `es` is the `e2eService` and `kubeletCmd` is the `*killCmd` for the kubelet service. 

Unfortunately, because the service failed to start, this pointer is nil, and we get a nil pointer dereference in the `Kill` function (in `test/e2e_node/e2e_service.go`). 

Off the top of my head, I see two potential solutions to this problem.

1. Pre-allocate default `killCmd` structs on the `e2eService` struct created in the `newE2eService` function in `test/e2e_node/e2e_service.go`. We could just make `killCmd.cmd = nil` on these, because the assumption appears to be that if `killCmd.cmd == nil`, the service is not running. (It appears that prior to the addition of `killCmd`, a nil `cmd` was interpreted as 'service not running,' and that this `cmd` field got rolled into `killCmd`, see 66d74f969dbbde894949b87366d215e020167faf).
2. Or, simply add an additional check to the `Kill` function, where we assume that if the `killCmd` is `nil`, the respective service is not running.

Thoughts? I want to make sure I'm not missing something here.

@timstclair because you added teardown code in 66d74f969dbbde894949b87366d215e020167faf
@euank because you added `killCmd` in 34996f1b84104a83033fb62cec72ec86f9eef8c1

Note: Tim is on vacation so I'm not really expecting a response from him.

### PR DESCRIPTION
**Pull request by mtaufen**
Fixes #27582.

We used to maintain a pointer variable for each process to kill after the
tests finish. @lavalamp suggested using a slice instead, which is a much
cleaner solution. This implements @lavalamp's suggestion and also extends
the idea to tracking directories that need to be removed after the tests finish.

This also means that we should no longer check for nil `killCmd`s inside
`func (k *killCmd) Kill() error {...}` (see #27582 and #27589). If a nil
`killCmd` makes it in there, something is bad elsewhere and we want to see
the nil pointer exception immediately.

Mentioning @timstclair and @euank wrt the original issue/PR.

### CODE REVIEW

File: test/e2e_node/e2e_service.go [hunk 1]
```diff
@@ -152,25 +154,15 @@ func isJournaldAvailable() bool {
...
 func (es *e2eService) stop() {
-	if err := es.stopService(es.kubeletCmd); err != nil {
-		glog.Errorf("Failed to stop kubelet: %v", err)
-	}
-	if es.kubeletStaticPodDir != "" {
-		err := os.RemoveAll(es.kubeletStaticPodDir)
-		if err != nil {
-			glog.Errorf("Failed to delete kubelet static pod directory %s.\n%v", es.kubeletStaticPodDir, err)
+	for _, k := range es.killCmds {
+		if err := es.stopService(k); err != nil {
```
Comment (timstclair) [inline test/e2e_node/e2e_service.go]: nit: delete stopService, just inline the call to kill

**Comment by timstclair** [comment]
LGTM

**Comment by euank** [comment]
LGTM as well

File: test/e2e_node/e2e_service.go (hunk 1 above)
Comment (mtaufen) [inline test/e2e_node/e2e_service.go]: Done

File: test/e2e_node/e2e_service.go [hunk 2]
```diff
@@ -319,19 +307,19 @@ type killCmd struct {
 }
 
 func (k *killCmd) Kill() error {
-	if k == nil {
-		glog.V(2).Infof("The killCmd is nil, nothing will be killed")
-		return nil
-	}
+
```
Comment (lavalamp) [inline test/e2e_node/e2e_service.go]: nit: extra blank line

**Comment by lavalamp** [comment]
LGTM. Removes more code than it adds, that's how you know it's a good generalization ;)

File: test/e2e_node/e2e_service.go (hunk 2 above)
Comment (mtaufen) [inline test/e2e_node/e2e_service.go]: Fixed.

**Comment by mtaufen** [comment]
Right on, thanks for your help Daniel. :)

**Comment by lavalamp** [comment]
Thanks again for the cleanup!

**Comment by mtaufen** [comment]
@k8s-bot test this

**Pull request merged**